components/pluralrules/src/operands.rs

```diff
@@ -202,48 +202,36 @@ impl_signed_integer_type!(i8 i16 i32 i64 i128 isize);
 
 impl From<&FixedDecimal> for PluralOperands {
     /// Converts a [icu_num_util::FixedDecimal] to [PluralOperands]
-    fn from(f: &FixedDecimal) -> Self {
-        let mag_range = f.magnitude_range();
+    fn from(dec: &FixedDecimal) -> Self {
+        let mag_range = dec.magnitude_range();
         let mag_high = *mag_range.end();
         let mag_low = *mag_range.start();
 
-        let mut integer_part: u64 = 0;
-        let integer_range = 0..=mag_high;
-        for magnitude in integer_range.rev() {
-            integer_part *= 10;
-            integer_part += f.digit_at(magnitude) as u64;
+        let mut i: u64 = 0;
+        for magnitude in (0..=mag_high).rev() {
+            i *= 10;
+            i += dec.digit_at(magnitude) as u64;
         }
 
-        // The fractional part of the number, expressed as an integer to preserve trailing zeroes.
-        // For example, could be "100" if the stored number is "0.100".
-        let mut fraction_part_full: u64 = 0;
-        // A running total of the number of trailing zeros seen in the fractional part of the
-        // number.
-        let mut num_trailing_zeros = 0;
-        // 10^num_trailing_zeros.
-        let mut weight_trailing_zeros = 1;
-        let fractional_magnitude_range = mag_low..=-1;
-        for magnitude in fractional_magnitude_range.rev() {
-            let digit = f.digit_at(magnitude);
-            if digit == 0 {
-                num_trailing_zeros += 1;
-                weight_trailing_zeros *= 10;
-            } else {
-                num_trailing_zeros = 0;
-                weight_trailing_zeros = 1;
+        let mut f: u64 = 0;
+        let mut t: u64 = 0;
+        let mut w: usize = 0;
+        for magnitude in (mag_low..=-1).rev() {
+            let digit = dec.digit_at(magnitude) as u64;
+            f *= 10;
+            f += digit;
+            if digit != 0 {
+                t = f;
+                w = (-magnitude) as usize;
             }
-            fraction_part_full *= 10;
-            fraction_part_full += digit as u64;
         }
-        let fraction_part_nozeros = fraction_part_full / weight_trailing_zeros;
-        let num_fractional_digits = (-std::cmp::min(0, mag_low)) as usize;
 
         PluralOperands {
-            i: integer_part,
-            v: num_fractional_digits,
-            w: num_fractional_digits - num_trailing_zeros,
-            f: fraction_part_full,
-            t: fraction_part_nozeros,
+            i,
+            v: (-mag_low) as usize,
```

Comment:
Is `mag_low <= 0`  guaranteed?

Reply:
Yes.  Good question.  From the [API docs](https://github.com/unicode-org/icu4x/blob/256eeeb342faec75c24026bbe65d8dce4bb8caab/components/num-util/src/fixed_decimal.rs#L217):

> Magnitude 0 is always included, even if the number has leading or trailing zeros.

This implies that the upper magnitude is always >= 0, and the lower magnitude is always <= 0.